pyseir/rt/infer_rt.py

```diff
@@ -0,0 +1,617 @@
+import math
+from datetime import timedelta
+import logging
+import structlog
+
+import numpy as np
+import pandas as pd
+from scipy import stats as sps
+
+from pyseir import load_data
+from pyseir.utils import TimeseriesType, get_run_artifact_path, RunArtifact
+from pyseir.rt.constants import InferRtConstants
+from pyseir.rt import plotting, utils
+
+rt_log = structlog.get_logger(__name__)
+
+
+def run_rt_for_fips(fips):
+    """Entry Point for Infer Rt"""
+
+    # Generate the Data Packet to Pass to RtInferenceEngine
+    input_df = _generate_input_data(fips, include_testing_correction=True)
+
+    # Save a reference to instantiated engine (eventually I want to pull out the figure generation
+    # and saving so that I don't have to pass a display_name and fips into the class
+    engine = RtInferenceEngine(
+        data=input_df, display_name=_get_display_name(fips), fips=fips, include_deaths=False
+    )
+
+    # Generate the output DataFrame (consider renaming the function infer_all to make it clearer)
+    output_df = engine.infer_all()
+
+    # Save the output to json for downstream repacking and incorporation.
+    county_output_file = get_run_artifact_path(fips, RunArtifact.RT_INFERENCE_RESULT)
+    if output_df is not None and not output_df.empty:
+        output_df.to_json(county_output_file)
+
+
+def _get_display_name(fips) -> str:
+    """Need to find the right function for this. Right now just return the fips"""
+    return str(fips)
+
+
+def _generate_input_data(fips, include_testing_correction=True, include_deaths=True):
+    """
+    Allow the RtInferenceEngine to be agnostic to aggregation level by handling the loading first
+
+    include_testing_correction: bool
+        If True, include a correction for testing increases and decreases.
+    """
+    times, observed_new_cases, observed_new_deaths = load_data.load_new_case_data_by_fips(
+        fips, t0=InferRtConstants.REF_DATE, include_testing_correction=include_testing_correction,
+    )
+
+    date = [InferRtConstants.REF_DATE + timedelta(days=int(t)) for t in times]
+
+    df = filter_and_smooth_input_data(
+        df=pd.DataFrame(dict(cases=observed_new_cases, deaths=observed_new_deaths), index=date),
+        include_deaths=include_deaths,
+    )
+    return df
+
+
+def filter_and_smooth_input_data(df: [pd.DataFrame], include_deaths=False) -> pd.DataFrame:
+    """Do Everything Strange Here Before it Gets to the Inference Engine"""
+    MIN_CUMULATIVE_COUNTS = dict(cases=20, deaths=10)
+    MIN_INCIDENT_COUNTS = dict(cases=5, deaths=5)
```

Comment:
I'd prefer if these weren't caps - I expect capitalized to be more global values - Is there a reason this doesn't belong in RtConstants?